internal/api/cluster_installation_test.go

```diff
@@ -554,3 +554,299 @@ func TestRunClusterInstallationMattermostCLI(t *testing.T) {
+		require.NoError(t, err)
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	})
+
+	t.Run("empty payload", func(t *testing.T) {
+		resp, err := http.Post(fmt.Sprintf("%s/api/cluster_installations/migrate", ts.URL), "application/json", bytes.NewReader([]byte("")))
+		require.NoError(t, err)
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+	})
+
+	client := model.NewClient(ts.URL)
+	t.Run("missing source cluster", func(t *testing.T) {
+		err := client.MigrateClusterInstallation(&model.MigrateClusterInstallationRequest{ClusterID: "", TargetCluster: "4567"})
+		require.EqualError(t, err, "failed with status code 400")
+	})
+
+	t.Run("missing target cluster", func(t *testing.T) {
+		err := client.MigrateClusterInstallation(&model.MigrateClusterInstallationRequest{ClusterID: "12345", TargetCluster: ""})
+		require.EqualError(t, err, "failed with status code 400")
+	})
+
+	t.Run("no cluster installation found to migrate", func(t *testing.T) {
+		err := client.MigrateClusterInstallation(&model.MigrateClusterInstallationRequest{ClusterID: "12345", TargetCluster: "67899"})
+		require.EqualError(t, err, "failed with status code 404")
+	})
+
+	// Valid migration test
+	installation1, err := client.CreateInstallation(&model.CreateInstallationRequest{
+		OwnerID:  "owner1",
+		Version:  "version",
+		Image:    "custom-image",
+		DNS:      "dns1.example.com",
+		Affinity: model.InstallationAffinityIsolated,
+	})
+	require.NoError(t, err)
+
+	time.Sleep(1 * time.Millisecond)
+
+	installation2, err := client.CreateInstallation(&model.CreateInstallationRequest{
+		OwnerID:  "owner2",
+		Version:  "version",
+		Image:    "custom-image",
+		DNS:      "dns2.example.com",
+		Affinity: model.InstallationAffinityIsolated,
+	})
+	require.NoError(t, err)
+	sourceClusterID := model.NewID()
+	targetClusterID := model.NewID()
+	clusterInstallation1 := &model.ClusterInstallation{
+		ClusterID:      sourceClusterID,
+		InstallationID: installation1.ID,
+		Namespace:      "namespace_10",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+
+	time.Sleep(1 * time.Millisecond)
+
+	clusterInstallation2 := &model.ClusterInstallation{
+		ClusterID:      sourceClusterID,
+		InstallationID: installation2.ID,
+		Namespace:      "namespace_11",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+
+	err = sqlStore.CreateClusterInstallation(clusterInstallation1)
+	require.NoError(t, err)
+
+	err = sqlStore.CreateClusterInstallation(clusterInstallation2)
+	require.NoError(t, err)
+
+	t.Run("valid migration test", func(t *testing.T) {
+		mcir := &model.MigrateClusterInstallationRequest{ClusterID: sourceClusterID, TargetCluster: targetClusterID, InstallationID: "", DNSSwitch: true, LockInstallation: true}
+		t.Log(mcir)
+		err := client.MigrateClusterInstallation(mcir)
+		require.NoError(t, err)
+	})
```

Comment:
This test actually causes server to panic as Cluster with such ID does not exist and the `nil pointer dereference` occurs. We should handle `nil` in the handler. The test succeeded because there is a bug in client method.

Reply:
Good point @Szymongib , made the required the changes. Thank you.